packages/lesswrong/components/ultraFeed/ultraFeedSettingsValidation.ts

```diff
@@ -1,56 +1,83 @@
 import { z } from 'zod';
 import { DEFAULT_SETTINGS } from './ultraFeedSettingsTypes';
+import type { FeedItemSourceType } from './ultraFeedTypes';
+import type { UltraFeedSettingsType } from './ultraFeedSettingsTypes';
 
-const ascendingNumbersOrNull = z.array(z.number().positive().nullable()).refine(arr => {
+const sourceWeightsSchema = z.record(
+  z.custom<FeedItemSourceType>(), 
```

Comment:
This is probably not what you want, as it doesn't perform any actual validation: https://zod.dev/?id=custom-schemas

Reply:
You probably want a zod enum defined from the runtime constituents of `FeedItemSourceType`, i.e. the combination of `feedPostSourceTypesArray`, `feedCommentSourceTypesArray`, and `feedSpotlightSourceTypesArray`.  (Or whatever it's updated to in this PR.)